sstables/sstables.hh

```diff
@@ -123,6 +123,14 @@ constexpr const char* snapshots_dir = "snapshots";
 constexpr const char* quarantine_dir = "quarantine";
 constexpr const char* pending_delete_dir = "pending_delete";
 
+constexpr auto table_subdirectories = std::to_array({
+    staging_dir,
+    upload_dir,
+    snapshots_dir,
+    quarantine_dir,
+    pending_delete_dir,
+});
+
```

Comment:
I made a comment about moving it out-of-line, but since the constituents are still inline, you can keep it like that for now and we'll clean everything up later.